src/components/exercises/VideoQuestion.js

```diff
@@ -0,0 +1,129 @@
+import React, {Component} from 'react';
+import {View, Text, TouchableWithoutFeedback, TouchableOpacity, Image} from "react-native";
+import {addIndex, map} from 'ramda';
+import {PRIMARY, GREEN} from '../../styles/colors'
+import {DEFAULT} from '../../styles/text'
+import {RoundTextButton} from '../Components';
+import {Video} from "expo";
+import playIcon from '../../assets/images/play.png';
+
+const replayIcon = playIcon;
+const mapIndexed = addIndex(map);
+
+const styles = {
+  answer: {
+    margin: 5
+  },
+  vidContainer: {
+    flex: 3,
+    alignItems: 'center',
+    justifyContent: 'center',
+  },
+  video: {
+    width: '90%',
+    height: '100%',
+  },
+  button: {
+    position: 'absolute',
+    padding: 15,
+    width: 80,
+    height: 80,
+    borderRadius: 9999,
+    backgroundColor: 'rgba(0, 0, 0, 0.6)',
+    zIndex: 9999
+  },
+  icon: {
+    width: 50,
+    height: 50
+  },
+  hidden: {
+    width: 0,
+    height: 0,
+    padding: 0,
+    overflow: 'hidden'
+  },
+  highlighted: {
+    backgroundColor: GREEN
+  }
+};
+
+class VideoQuestion extends Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      highlighted: null,
+      isPaused: false,
+      isFinished: false
+    };
+  }
+
+  onEnd = () => {
+    this.setState({
+      isFinished: true
+    });
+  };
+
+  playPause = () => {
+    this.setState({
+      isPaused: !this.state.isPaused
+    });
+    if (!this.state.isPaused) {
+      this.setState({isFinished: false})
+    }
+  };
+
+  replay = () => {
+    this.player.seek(0);
+    this.setState({
+      isPaused: false,
+      isFinished: false
+    });
+  };
+
+  selectAnswer = key => () => {
+    this.setState(({highlighted}) => {
+      highlighted = highlighted === key ? null : key;
```

Comment:
You overwrite the parameter here. That's bad practice in general. You could instead do:

```.js
this.setState({ highlighted }) => ({
  highlighted: highlighted === key ? null : key
})
```